src/lib/components/buttons/file-chooser/terra-file-chooser.component.ts

```diff
@@ -76,62 +60,41 @@ export class TerraFileChooserComponent extends TerraButtonComponent {
         TerraFileBrowserComponent
     >();
 
-    @ViewChild('overlay', { static: true })
-    public overlay: TerraOverlayComponent;
-
-    @ViewChild('fileBrowser', { static: true })
+    @ViewChild('fileBrowser', { static: false })
     public fileBrowser: TerraFileBrowserComponent;
 
-    public primaryOverlayButton: TerraOverlayButtonInterface;
+    @ViewChild(TemplateRef, { static: true })
+    public dialogTemplateRef: TemplateRef<any>;
 
-    public secondaryOverlayButton: TerraOverlayButtonInterface;
+    public _translationPrefix: string = 'terraFileInput';
 
-    private _translationPrefix: string = 'terraFileInput';
+    public _selectedObject: TerraStorageObject;
 
     private _primaryBrowserButtonCaption: string = '';
 
     private _secondaryBrowserButtonCaption: string = '';
 
-    private _selectedObject: TerraStorageObject;
-
-    private _storageServices: Array<TerraBaseStorageService>;
-
-    constructor(private _translation: L10nTranslationService) {
+    constructor(private _translation: L10nTranslationService, private _dialog: MatDialog) {
         super();
-
-        this.primaryOverlayButton = {
-            icon: 'icon-success',
-            caption: this.inputPrimaryBrowserButtonCaption,
-            isDisabled: true,
-            clickFunction: (): void => {
-                this.outputSelected.emit(this._selectedObject);
-                this.overlay.hideOverlay();
-            }
-        };
-
-        this.secondaryOverlayButton = {
-            icon: 'icon-close',
-            caption: this.inputSecondaryBrowserButtonCaption,
-            isDisabled: false,
-            clickFunction: (): void => {
-                this.outputCancelled.emit();
-                this.overlay.hideOverlay();
-            }
-        };
```

Comment:
theoretically we would be able to remove these properties, since it would be a breaking change.. :/

Reply:
nope. just the properties.